src/metabase/driver/clickhouse.clj

```diff
@@ -157,8 +163,20 @@
 
 (defmethod driver/create-table! :clickhouse
   [driver db-id table-name column-definitions & {:keys [primary-key]}]
-  (let [sql (create-table!-sql driver table-name column-definitions :primary-key primary-key)]
-    (qp.writeback/execute-write-sql! db-id sql)))
+  (sql-jdbc.execute/do-with-connection-with-options
+   driver
+   db-id
+   {:write? true}
+   (fn [^java.sql.Connection conn]
+     (if (cloud? conn)
+       (throw (Exception. "Only supported for ClickHouse Cloud"))
+       (with-open [stmt (.createStatement conn)]
+         (let [stmt    (.unwrap stmt ClickHouseStatementImpl)
+               request (.getRequest stmt)]
+           (.set request "wait_end_of_query" 1)
+           (with-open [_response (-> request
+                                     (.query (create-table!-sql driver table-name column-definitions :primary-key primary-key))
+                                     (.executeAndWait))])))))))
```

Comment:
@slvrtrn Rather than set `wait_end_of_query` on the connection string, which I think would require changes to `do-with-connection-with-options`, I opted to use the client API to achieve the same goal.

I drew inpsiration from this example:
https://github.com/ClickHouse/clickhouse-java/blob/6a4ffc4ae0b76a6f65a57ee0db3246d45fbc78c1/examples/jdbc/src/main/java/com/clickhouse/examples/jdbc/Advanced.java#L150-L170

Can you confirm whether this works for you?

Reply:
Yes, it should also work this way. You can set it on the connection level or per individual query.